dart-impl/mpi/src/dart_team_group.c

```diff
@@ -27,9 +27,10 @@
 
 static struct dart_group_struct* allocate_group()
 {
-  struct dart_group_struct* group = malloc(sizeof(struct dart_group_struct));
+  struct dart_group_struct* group = (struct dart_group_struct *) malloc(
```

Comment:
> Unix API Allocator sizeof operand mismatch

I couldn't immediately find this warning in your _before_ analyzer output. However, I think what it refers to is a (potential) difference in the sizes of the type used in `sizeof` and the assigned type (such `dart_unit_t` vs `dart_global_unit_t`). So the cast actually hides that. 

> Does this hurt?

I think it definitely hurts readability as it clutters the code even more, unnecessarily so, and repeats the type three times in a single statement. On the positive side, it might trigger a warning if the target type changes but the cast does not. A clean solution could be:
```
struct dart_group_struct* group = malloc(sizeof(*group));
```
which would account for any changes in the type of `group`. 

I agree that we have to be consistent on this throughout DART. I have been removing the casts in pretty much all the code I touched but haven't thought about the solution outlined above, which I got from http://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc